tests/Rules/PHPUnit/data/assert-same.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace PHPUnit;
+
+class FooTestCase extends \PHPUnit\Framework\TestCase
+{
+
+	public function testAssertSame()
+	{
+		$this->assertSame('1', 1);
+		$this->assertSame('1', new \stdClass());
+		$this->assertSame(1, $this->returnsString());
+		$this->assertSame('1', self::returnsInt());
```

Comment:
There should also be a few examples of positive cases - comparing int/int and probably also comparing some unions against simple type and unions against unions.

Reply:
added bellow